subiquity/server/controllers/filesystem.py

```diff
@@ -376,7 +371,8 @@ async def v2_add_partition_POST(self, data: AddPartitionV2) \
             'mount': data.partition.mount,
         }
 
-        self.create_partition(disk, data.gap, spec, wipe='superblock')
+        gap = gaps.at_offset(disk, data.gap.offset).split(requested_size)[0]
```

Comment:
This translation between types.Gap and gaps.Gap is, hmm. A thing :) I guess it's needed but it feels a bit silly somehow.

Reply:
Yes, the same-sounding types between what is provided by common.types and what filesystem actually uses is a bit unfortunate.  If we can come up with an answer for clarity there in general it will also probably help this case.